src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,984 @@
+The Storage and GraphQL API categories can be used together to associate a file, such as an image or video, with a particular record. For example, you might create a `Person` model with a profile picture, or a `Post` model with an associated image. With Amplify's GraphQL API and Storage categories, you can reference the file within the model itself to create an association.
+
+To get started, run the following in an existing Amplify project:
+
+```bash
+npm i @aws-amplify/ui-react  #For Authenticator component
+amplify add auth             #Select default configuration
+amplify add storage          #Select S3 with read/write access
+amplify add api              #Select Cognito User Pool for authorization type
+```
+
+When prompted, use the following schema:
+
+```graphql
+type Song @model {
+  id: ID!
+  name: String!
+  covertArtKey: String
+}
+```
+
+Save the schema and run `amplify push` to deploy the changes.
+
+To use files and attachments you need authorization credentials for reading and writing to Amazon S3, which `amplify add auth` configures in the default setting along with a Cognito User Pool. For more details on Storage / API authorization, see the [Configuring Authorization](#configuring-authorization) section below.
+
+For the complete working example, including required imports and React component state management, see the [Complete Example](#complete-example) below.
+
+## Creating a record with a GraphQL API
+
+Running the following mutation will create a record with the GraphQL API. The corresponding file will be associated with this record.
```

Comment:
> The corresponding file will be associated with this record.

In the code snippets below you don't talk about the file just yet. So I think there's a bit of a disconnect here for the reader.